### PR TITLE
Removing duplicate callback to create auth token

### LINKED ISSUE #2620: Two access tokens created for each new user
**Describe the bug**

When a new user is created, a single access token should be created.

**To Reproduce**

```
# app/models/user.rb:88
after_create_commit :create_access_token
```
```
# app/models/concerns/access_tokenable.rb:5
after_create :create_access_token
```

**Expected behavior**

One access token should be created when a user is created.

**Screenshots**

image

### PR DESCRIPTION
**Pull request by mrunalselokar**
## Description
Removed after create callback in user model to create a new access token. This is already being taken care of by  `AccessTokenable` module.

Fixes #2620 
## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

Updated `user_spec.rb` to verify that only a single auth token is created for a newly created user.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by pranavrajs** [comment]
@mrunalselokar Thanks for the PR 👍. 

@sojan-official Please take a look at this and see if we can get it merged soon.

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -72,4 +72,12 @@
       expect(user.valid_sso_auth_token?(sso_auth_token)).to eq false
     end
   end
+
+  describe 'access token' do
+    it 'creates a single access token upon user creation' do
+      new_user = create(:user)
+      token_count = AccessToken.where(owner_id: new_user.id).count
```
Comment (sojan-official) [inline spec/models/user_spec.rb:79]:
```suggestion
      token_count = AccessToken.where(owner: new_user).count
```

`owner` attribute is a polymorphic column. so we should pass both `owner_id` and `owner_type`, or pass the object and rails takes care of it for us.

**Review by sojan-official — CHANGES_REQUESTED** [review]
@mrunalselokar  Thanks for the changes.  I have added a minor comment on the specs. 
Also could you add a data migration for removing the duplicate access tokens for the existing user account ? 

Let me know if you need help with that.

**Comment by sojan-official** [comment]
@mrunalselokar  Thanks for the quick updates. The changes look good to me. We will test it against a staging database and get this merged.

**Review by sojan-official — APPROVED** [review]

**Review by muhsin-k — APPROVED** [review]

**Comment by muhsin-k** [comment]
@sojan-official QA has been done on staging and everything is working as expected.

**Pull request merged**